tests/pkg_fatfs_vfs/main.c

```diff
@@ -295,6 +295,61 @@ static void test_create(void)
     print_test_result("test_create__umount", vfs_umount(&_test_vfs_mount) == 0);
 }
 
+#ifdef MODULE_NEWLIB
+static void test_newlib(void)
+{
+    FILE* fl;
+    char buf[sizeof(test_txt) + sizeof(test_txt2)];
+    print_test_result("test_newlib__mount", vfs_mount(&_test_vfs_mount) == 0);
+
+    /* try to open file that doesn't exist */
+    fl = fopen(FULL_FNAME_NXIST, "r");
+    print_test_result("test_newlib__fopen", fl == NULL);
+    if (fl) {
+        fclose(fl);
+    }
+
+    /* open file RO */
+    fl = fopen(FULL_FNAME1, "r");
+    print_test_result("test_newlib__fopen_r", fl != NULL);
```

Comment:
This will always fail as that file got renamed by `test_rename()` - and now `test_create()` fails.

Reply:
Ah, true! I was concentrating on the other tests, i'll just move this one down after the write test.

Reply:
Well that failure in `test_create` is new, doesn't happen on `master`

Reply:
Hmm, i'm wondering if in `test_create` `vfs_open` with ` O_WRONLY | O_CREAT` should be successful if  the file already exists? Shouldn't there be `O_TRUNC` added?

Reply:
Why wouldn't it be successful? 
`O_WRONLY | O_CREAT` means "Open file for writing, create it if it doesn't exist". 
So this should definitely work.

Reply:
Ok, i agree. Before `O_WRONLY | O_CREAT` was translated to `FA_CREATE_NEW`, so if the file exists it failed. I'll add a test for this.